vm-deployment/vm.bicep

```
@@ -0,0 +1,218 @@
+  location: location
+  sku: {
+    name: 'Standard'
+    tier: 'Regional'
+  }
+  properties: {
+    publicIPAddressVersion: 'IPv4'
+    publicIPAllocationMethod: 'Static'
+    ddosSettings: {
+      protectionMode: 'VirtualNetworkInherited'
+    }
+  }
+}
+
+
+// Virtual network
+resource virtualNetwork 'Microsoft.Network/virtualNetworks@2023-11-01' = {
+  name: virtualNetworkName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        vnetAddressPrefix
+      ]
+    }
+    subnets: [
+      {
+        name: 'default'
+        properties: {
+          addressPrefix: subnetAddressPrefix
+          privateEndpointNetworkPolicies: 'Disabled'
+          privateLinkServiceNetworkPolicies: 'Enabled'
+        }
+      }
+    ]
+  }
+}
+
+
+
+// Virtual machine
+resource virtualMachine 'Microsoft.Compute/virtualMachines@2024-03-01' = {
+  name: virtualMachineName
+  location: location
+  properties: {
+    hardwareProfile: {
+      vmSize: vmSize
+    }
+    priority: vmPriority
+    storageProfile: {
+      imageReference: {
+        publisher: 'Canonical'
+        offer: '0001-com-ubuntu-server-jammy'
+        sku: '22_04-lts-gen2'
+        version: 'latest'
+      }
+      osDisk: {
+        osType: 'Linux'
+        name: '${virtualMachineName}_OsDisk_1'
+        createOption: 'FromImage'
+        managedDisk: {
+          storageAccountType: diskSku
+        }
+        diskSizeGB: diskSizeGB
+      }
+    }
+    osProfile: {
+      computerName: virtualMachineName
+      adminUsername: adminUsername
+      linuxConfiguration: {
+        disablePasswordAuthentication: true
+        ssh: {
+          publicKeys: [
+            {
+              path: '/home/${adminUsername}/.ssh/authorized_keys'
+              keyData: sshPublicKey
+            }
+          ]
+        }
+      }
+    }
+    networkProfile: {
+      networkInterfaces: [
+        {
+          id: networkInterfaceResource.id
+        }
+      ]
+    }
+  }
+}
+// Network interface
+resource networkInterfaceResource 'Microsoft.Network/networkInterfaces@2023-11-01' = {
+  name: networkInterfaceName
+  location: location
+  properties: {
+    ipConfigurations: [
+      {
+        name: 'ipconfig1'
+        properties: {
+          privateIPAddress: privateIPAddress
+          publicIPAddress: {
+            id: publicIPAddressResource.id
+            properties: {
+              deleteOption: 'Detach'
+            }
+          }
+          subnet: {
+            id: '${virtualNetwork.id}/subnets/default'
+          }
+          primary: true
+        }
+      }
+    ]
+    networkSecurityGroup: {
+      id: networkSecurityGroup.id
+    }
+    // The auxiliaryMode, auxiliarySku and disableTcpStateTracking values are set to their defaults to prevent
+    // `what-if` command noise.
+    // See https://github.com/Azure/arm-template-whatif/issues/356
+    disableTcpStateTracking: false
+    auxiliaryMode: 'None'
+    auxiliarySku: 'None'
+  }
+}
```

```suggestion
}

output sshCommand string = 'ssh ${adminUsername}@${publicIPAddress.properties.ipAddress}'
```
Could we add an output to guide the user on how to connect to the VM?